ibmwatson.class.nut

```diff
@@ -0,0 +1,263 @@
+class IBMWatson {
+
+    static version = [1, 0, 0];
+
+    static INVALID_REQUEST_ERROR = "Error: Invalid Request";
+    static INVALID_AUTH_TOKEN_ERROR = "Error: Invalid Authentication Token";
+    static INVALID_AUTH_METHOD_ERROR = "Error: Invalid API Key or Authentication Method";
+    static MISSING_RESOURCE_ERROR = "Error: Resource does not exist";
+    static CONFLICT_ERROR = "Error: Conflict, resource already exists";
+    static UNEXPECTED_ERROR = "Error: Unexpected error";
+
+    _apiKey = null;
+    _authToken = null;
+    _baseURL = null;
+
+    /***************************************************************************************
+     * Constructor
+     * Returns: null
+     * Parameters:
+     *      apiKey : string - Watson API key (generate via Accees -> API Keys)
+     *      authToken : string - Watson Authorization Token (provided when API Key is created)
+     *      orgID : string - Organization ID
+     *      version (optional) : string - Watson API version number
+     **************************************************************************************/
+    constructor(apiKey, authToken, orgID, version = "v0002") {
+        _apiKey = apiKey;
+        _authToken = authToken;
+
+        local protocol = (orgID == "quickstart") ? "http" : "https";
+        _baseURL = format("%s://%s.internetofthings.ibmcloud.com/api/%s", protocol, orgID, version);
+    }
+
+    /***************************************************************************************
+     * postData
+     * Returns: null
+     * Parameters:
+     *      typeID : string - device type ID
+     *      deviceID : string - ID of device that is posting data
+     *      eventID : string - ID to identify event datastream
+     *      data : table - event data (formatted: {"d" : "data here", "ts": "valid ISO8601 timestamp" })
+     *      headers (optional) : table - additional headers to add to the http request
+     *      cb (optional) : function - function to execute when response received
+     **************************************************************************************/
+    function postData(typeID, deviceID, eventID, data, headers = {}, cb = null) {
+        // POST "device/types/${typeId}/devices/${deviceId}/events/${eventId}"
+        if (typeof headers == "function") {
```

Comment:
My concern is that we have this "conversion" in every single method, which blows up the code. So may be in some cases (I'm not sure how critical is that for this particular library) to make people passing extra Null than have some memory used for the sake of their convenience.